File: src/main/java/com/dems/dashboard/DashboardService.java
package com.dems.dashboard;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * Service interface aggregating enterprise-grade read-only dashboard metrics and analytics across DEMS repositories.
 */
public interface DashboardService {

    DashboardSummaryResponse getSummary(OffsetDateTime startDate, OffsetDateTime endDate);

    CaseAnalyticsResponse getCaseAnalytics(OffsetDateTime startDate, OffsetDateTime endDate);

    EvidenceAnalyticsResponse getEvidenceAnalytics(OffsetDateTime startDate, OffsetDateTime endDate);

    IntegrityAnalyticsResponse getIntegrityAnalytics(OffsetDateTime startDate, OffsetDateTime endDate);

    CustodyAnalyticsResponse getCustodyAnalytics(OffsetDateTime startDate, OffsetDateTime endDate);

    AuditAnalyticsResponse getAuditAnalytics(OffsetDateTime startDate, OffsetDateTime endDate);

    List<RecentActivityDto> getRecentActivities();

    SystemHealthResponse getSystemHealth();
}

File: frontend/src/services/dashboardService.js
import apiClient from './apiClient';
import { API_ENDPOINTS } from '../constants/apiEndpoints';

/**
 * Service for fetching executive dashboard and module analytics from Spring Boot DashboardController.
 */
export const dashboardService = {
  getSummary: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.SUMMARY, { params });
  },

  getCaseAnalytics: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.CASES, { params });
  },

  getEvidenceAnalytics: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.EVIDENCE, { params });
  },

  getIntegrityAnalytics: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.INTEGRITY, { params });
  },

  getCustodyAnalytics: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.CUSTODY, { params });
  },

  getAuditAnalytics: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.AUDIT, { params });
  },

  getRecentActivities: async () => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.RECENT_ACTIVITIES);
  },

  getSystemHealth: async () => {
    return await apiClient.get(API_ENDPOINTS.DASHBOARD.SYSTEM_HEALTH);
  },
};

export default dashboardService;
